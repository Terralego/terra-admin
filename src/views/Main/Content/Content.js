import React, { lazy, Suspense } from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import Loading from '../../../components/Loading';

const Summary = lazy(() => import('../../Summary'));

export const Content = ({ modules, defaultRoute }) => (
  <main className="main-content">
    <Switch>
      {modules.map(Component => (
        <Route
          key={Component.config.path}
          path={Component.config.path}
        >
          <Component />
        </Route>
      ))}
      {typeof defaultRoute === 'string' && <Redirect exact from="/" to={defaultRoute} />}
      <Route path="">
        <Suspense fallback={<Loading />}>
          <Summary />
        </Suspense>
      </Route>
    </Switch>
  </main>
);

export default Content;
