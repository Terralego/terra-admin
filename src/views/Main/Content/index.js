import { connectAuthProvider } from '@terralego/core/modules/Auth';
import { connectAppProvider } from '../../../components/AppProvider';

import { Content } from './Content';
import { getComponentsByEnabledModules } from '../../../services/modules';
import compose from '../../../utils/compose';


const getDefaultRoute = landingModule => {
  const [{ config: { path = '' } = {} } = {}] = getComponentsByEnabledModules([landingModule]);
  return path;
};

const setLandingModule = ({
  env: { landing_module: landingModule },
}) => ({
  defaultRoute: getDefaultRoute(landingModule),
  landingModule,
});

const componentsToDisplay = ({ user }) => {
  if (user) {
    const { modules } = user;
    return { modules: getComponentsByEnabledModules(modules) };
  }
  return { modules: [] };
};

export default compose(
  connectAppProvider(setLandingModule),
  connectAuthProvider(componentsToDisplay),
)(Content);
