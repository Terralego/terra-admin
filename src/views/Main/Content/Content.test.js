import React from 'react';
import renderer from 'react-test-renderer';
import { Content } from './Content';


jest.mock('react-router-dom', () => ({
  Route: ({ children }) => children,
  Switch: ({ children }) => children,
}));
jest.mock('../../Summary', () => () => <div>Summary</div>);
jest.mock('../../../components/Loading', () => () => <div>Loading</div>);

function Foo () { return <p>Module Foo</p>; }
Foo.config = {
  path: '/foo',
};
function Bar () { return <p>Module Bar</p>; }
Bar.config = {
  path: '/bar',
};
it('should render correctly', () => {
  const tree = renderer.create(
    <Content modules={[Foo, Bar]} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
