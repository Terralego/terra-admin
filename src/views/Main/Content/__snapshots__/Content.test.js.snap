// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<main
  className="main-content"
>
  <p>
    Module Foo
  </p>
  <p>
    Module Bar
  </p>
  <div>
    Loading
  </div>
</main>
`;
